File: Behavioral/Strategy/Cpp/PayPalPayment.cpp
// PayPalPayment.cpp

#include "PayPalPayment.h"

PayPalPayment::PayPalPayment(const std::string& email) : email(email) {}

void PayPalPayment::pay(double amount) {
  std::cout << "Paid $" << amount << " using PayPal (Email: " << email << ")" << std::endl;
}

File: Behavioral/Strategy/CSharp/PayPalPayment.cs
// PayPalPayment.cs

using System;

public class PayPalPayment : IPaymentStrategy {
  private string _email;

  public PayPalPayment(string email) {
    _email = email;
  }

  public void Pay(double amount) {
    Console.WriteLine($"Paid ${amount} using PayPal (Email: {_email})");
  }
}
